packages/core/src/core/prompts.ts

```diff
@@ -271,3 +271,81 @@ Your core function is efficient and safe assistance. Balance extreme conciseness
 
   return `${basePrompt}${memorySuffix}`;
 }
+
+/**
+ * Provides the system prompt for the state snapshotting process.
+ * This prompt instructs the model to act as a specialized state manager,
+ * think in a scratchpad, and produce a structured XML summary.
+ */
+export function getCompressionPrompt(): string {
+  return `
+You are the internal monologue and state management component of a software engineering AI agent. Your primary function is to perform "state snapshotting".
+
+When the conversation history grows too large, you will be invoked to distill the entire history into a concise, structured XML snapshot. This snapshot is CRITICAL, as it will become the agent's *only* memory of the past. The agent will resume its work based solely on this snapshot. All crucial details, plans, errors, and user directives MUST be preserved.
```

Comment:
"This snapshot is CRITICAL, as it will become the agent's *only* memory of the past" -> This will change with the recency bias PR. Can we remove or reword so that we don't mention what the summarization covers?